tracing-opentelemetry/src/layer.rs

```diff
@@ -387,12 +415,17 @@ where
         let span = ctx.span(id).expect("Span not found, this is a bug");
         let mut extensions = span.extensions_mut();
 
-        let mut builder = self
-            .tracer
-            .span_builder(attrs.metadata().name())
-            .with_start_time(SystemTime::now())
-            // Eagerly assign span id so children have stable parent id
-            .with_span_id(self.id_generator.new_span_id());
+        let mut span_name = None;
+        attrs.record(&mut SpanNameVisitor(&mut span_name));
```

Comment:
Instead of visiting all the fields once to find the name override, and then again to record the otel attributes, would it make more sense to just visit everything in one pass? Could we construct the builder with the name from the metadata, and then, if we find an `otel.name` field while recording the fields, replace it with that value?

I suppose the downside to this would be that we might allocate an additional string that we end up throwing away, but we wouldn't need to iterate over the attributes twice. I don't know off the top of my head which is more efficient...

Reply:
Yeah the perf is likely similar in common cases, but the code is simpler without the second pass over the attributes so that's probably a better way to go for now, updated.